ocaml/quicktest/qt_filter.ml

```diff
@@ -116,19 +116,41 @@ module SR = struct
     ; required_sm_api_version
     }
 
-  let list_selected_srs rpc session_id =
-    Client.Client.SR.get_all rpc session_id
-    |> List.filter
-      (fun sr ->
-         Client.Client.SR.get_PBDs rpc session_id sr
-         |> List.exists
-           (fun pbd -> Client.Client.PBD.get_currently_attached rpc session_id pbd)
-      )
-    |> List.map (get_sr_info rpc session_id)
+  let list_srs_connected_to_localhost rpc session_id =
+    let is_shared_or_local sr =
+      (** allows us to ignore srs which don't belong to us.
+        * this remedies some failing tests if we run the quicktest binary on a host in a pool *)
+      let pbds = Client.Client.SR.get_PBDs ~rpc ~session_id ~self:sr in
+      let does_pbd_belong_to_me pbd =
+        let pbd_host = Client.Client.PBD.get_host ~rpc ~session_id ~self:pbd in
+        let pbd_host_uuid = Client.Client.Host.get_uuid ~rpc ~session_id ~self:pbd_host in
+        pbd_host_uuid = A.localhost_uuid
+      in
+      match pbds with
+      | [pbd] -> (* sr is not shared, test it only if it is ours *)
+        does_pbd_belong_to_me pbd
+      | []    -> Printf.eprintf "SR (%s) has no pbds %s" (Ref.string_of sr) __LOC__; false
+      | _::_ as pbds  -> (** sr is shared, so we have access to is
+                           * do a sanity check that we own one of the pbds *)
+        if List.exists does_pbd_belong_to_me pbds then
+          true
+        else begin
+          Printf.eprintf "SR (%s) is shared, but we do not own any of its pbds... %s" (Ref.string_of sr) __LOC__;
+          false
+        end
+    in
+    let has_at_least_one_attached_pbd sr =
```

Comment:
this is wrong - need to make sure that sr's pbd that is ours is attached, not simply that any pbd is attached